appengine/queue/taskqueue.py

```diff
@@ -0,0 +1,449 @@
+
+    def maybe_insert_into_pq(self):
+        """
+        We only want to have items in the priority queue
+        which can be leased, so we check that dependencies
+        are met before inserting.
+
+        We use the time of insertion as the priority.
+        Which will be similar to enqueueTimestamp when
+        first inseting a task for first time. (with met dependencies)
+
+        And similar to leaseTimeStamp when retrying a task.
+        This means that when a task is retried it get pushed to the end
+        of the queue. If we would insert with enqueueTimestamp after a 
+        retry it would get executed inmediately.
+        """
+        if not self._children and not self._in_pq:
+            self._in_pq = True
+            self._queue.pq.add_item(now() , self.id)
+
+
+    def _set_retry_timer(self, lease_for):
+        """
+        Re insert in pq after certain time
+        """
+        Timer(lease_for, self.maybe_insert_into_pq, ()).start()
+ 
+    def lease(self, lease_for):
+        """
+        time to lease for in seconds
+        """
+        assert self._leaseTimeStamp <= now()
+        self._leaseTimeStamp = now() + lease_for * 10**6
+        self._set_retry_timer(lease_for)
+
+    def __hash__(self):
+        return hash(self.id)
+
+
+class TaskHandler(webapp2.RequestHandler):
+
+
+    def get(self, project_name, taskqueue_name, maybe_task_name):
+        queue =  queues[taskqueue_name]
+        if not maybe_task_name:
+            self._list_task(queue)
+        else:
+            self._get_task(queue, maybe_task_name)
+
+
+    def post(self, project_name, taskqueue_name, maybe_lease_or_task_name):
+        queue =  queues[taskqueue_name]
+        if maybe_lease_or_task_name == 'lease':
+            numTasks = int(self.request.get('numTasks', default_value='1'))
+            assert numTasks >= 1
+            leaseSecs = int(self.request.get('leaseSecs', default_value='1'))
+            assert leaseSecs >= 1
+            tag = self.request.get('tag', default_value='')
+
+            self._lease_task(queue, numTasks, leaseSecs, tag)
+        elif maybe_lease_or_task_name == '':
+            self._insert_task(queue, taskqueue_name)
+        else:
+            newLeaseSeconds = int(self.request.get('newLeaseSeconds', default_value='1'))
+            assert newLeaseSeconds >= 1
+            self._update_task(queue, maybe_lease_or_task_name, newLeaseSeconds)
+
+    def patch(self, project_name, taskqueue_name, task_name):
+        queue =  queues[taskqueue_name]
+        newLeaseSeconds = int(self.request.get('newLeaseSeconds', default_value='1'))
+        assert newLeaseSeconds >= 1
+        self._patch_task(queue, task_name, newLeaseSeconds)
+
+    def delete(self, project_name, taskqueue_name, task_name):
+        queue =  queues[taskqueue_name]
+        self._delete_task(queue, task_name)
+
+    def _delete_task(self, queue, task_name):
+        """
+        DELETE  /projectname/taskqueue/queuename/tasks/taskname
```

Comment:
Can you emphasize the parts of the call the user has to fill in? Maybe `/${projectname}/taskqueue/${queuename}/tasks/${taskname}` or something like that? Same for all the other requests.